app/routes/_gcn.circulars.$circularId.($version)/route.tsx

```diff
@@ -0,0 +1,185 @@
+import type { BreadcrumbHandle } from '~/root/Title'
+
+export const handle: BreadcrumbHandle<typeof loader> = {
+  breadcrumb({ data }) {
+    if (data) {
+      const { subject, version } = data
+      return `${version ? `v${version} - ` : ''} ${subject}`
+    }
+  },
+}
+
+export async function loader({
+  params: { circularId, version },
+}: LoaderFunctionArgs) {
+  if (!feature('CIRCULAR_VERSIONS') && version)
+    throw new Response(null, { status: 404 })
+  invariant(circularId)
+  const result = await get(
+    parseFloat(circularId),
+    version ? parseFloat(version) : undefined
+  )
+
+  return json(result, {
+    headers: {
+      ...publicStaticShortTermCacheControlHeaders,
+      ...getCanonicalUrlHeaders(new URL(`/circulars/${circularId}`, origin)),
+    },
+  })
+}
+
+export const headers: HeadersFunction = ({ loaderHeaders }) =>
+  pickHeaders(loaderHeaders, ['Link'])
+
+export default function () {
+  const { circularId, body, bibcode, version, ...frontMatter } =
+    useLoaderData<typeof loader>()
+  const searchString = useSearchString()
+
+  const result = useRouteLoaderData<typeof parentLoader>(
+    'routes/_gcn.circulars.$circularId'
+  )
+
+  const latest = version && !result?.history.includes(version)
+  const linkString = `/circulars/${circularId}${
+    !latest && version ? `/${version}` : ''
+  }`
+  return (
+    <>
+      <ButtonGroup>
+        <Link to={`/circulars${searchString}`} className="usa-button">
+          <div className="position-relative">
+            <Icon.ArrowBack
+              role="presentation"
+              className="position-absolute top-0 left-0"
+            />
+          </div>
+          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Back
+        </Link>
+        <ButtonGroup type="segmented">
+          <Link
+            to={`${linkString}.txt`}
+            className="usa-button usa-button--outline"
+            reloadDocument
+          >
+            Text
+          </Link>
+          <Link
+            to={`${linkString}.json`}
+            className="usa-button usa-button--outline"
+            reloadDocument
+          >
+            JSON
+          </Link>
+          {bibcode ? (
+            <Link
+              to={`https://ui.adsabs.harvard.edu/abs/${bibcode}`}
+              className="usa-button usa-button--outline"
+            >
+              Cite (ADS)
+            </Link>
+          ) : (
+            <Button
+              type="button"
+              disabled
+              outline
+              title="The ADS entry for this Circular is not yet available. ADS entries are updated every week on Monday at 08:00 UTC. Please check back later."
+            >
+              Cite (ADS)
+            </Button>
+          )}
+        </ButtonGroup>
+        {useFeature('CIRCULAR_VERSIONS') &&
+          result?.history &&
+          result?.history.length > 0 && (
+            <CircularsHistory circular={circularId} history={result?.history} />
+          )}
```

Comment:
Does TypeScript like this shorter version?

```suggestion
        {useFeature('CIRCULAR_VERSIONS') &&
          result?.history.length && (
            <CircularsHistory circular={circularId} history={result.history} />
          )}
```

Reply:
Unfortunately no, 
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/b9b09f93-9e8f-498f-a045-bbb08b17e4ed)